### PR TITLE
Log deprecations for old properties in `base.*` plugin extension

### PR DESCRIPTION
**Pull request by eskatos**
* Fixes https://github.com/gradle/gradle/issues/22908

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test AST

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test AST

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractKotlinPluginAndroidSmokeTest.groovy [hunk 1]
```diff
@@ -83,7 +84,7 @@ abstract class AbstractKotlinPluginAndroidSmokeTest extends AbstractSmokeTest im
         [kotlinPluginVersion, androidPluginVersion, parallelTasksInProject] << [
                 TestedVersions.kotlin.versions,
                 TestedVersions.androidGradle.versions,
-                ParallelTasksInProject.values(),
+                ParallelTasksInProject.values().take(1),
```
Comment (tresat) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractKotlinPluginAndroidSmokeTest.groovy:87]: ❓ Why this?  Shouldn't you just reference the enum token you want by name here?

**Review by tresat — APPROVED** [review]

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractKotlinPluginAndroidSmokeTest.groovy (hunk 1 above)
Comment (eskatos) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractKotlinPluginAndroidSmokeTest.groovy]: Good catch. It's a local only edit that shouldn't have made it in the PR. Reverted!

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**